patches/net/minecraft/network/syncher/EntityDataSerializers.java.patch

```diff
@@ -1,11 +1,22 @@
 --- a/net/minecraft/network/syncher/EntityDataSerializers.java
 +++ b/net/minecraft/network/syncher/EntityDataSerializers.java
-@@ -157,16 +_,16 @@
+@@ -156,17 +_,27 @@
+         FriendlyByteBuf::writeQuaternion, FriendlyByteBuf::readQuaternion
      );
  
++    private static final org.slf4j.Logger LOGGER = com.mojang.logging.LogUtils.getLogger();
++    /**
++     * @deprecated NeoForge: Use {@link net.neoforged.neoforge.registries.NeoForgeRegistries#ENTITY_DATA_SERIALIZERS} instead
++     */
++    @Deprecated
      public static void registerSerializer(EntityDataSerializer<?> p_135051_) {
 -        SERIALIZERS.add(p_135051_);
-+        if (SERIALIZERS.add(p_135051_) >= 256) throw new RuntimeException("Vanilla DataSerializer ID limit exceeded");
++        if (!StackWalker.getInstance(StackWalker.Option.RETAIN_CLASS_REFERENCE).getCallerClass().equals(EntityDataSerializers.class)) {
```

Comment:
Should the `StackWalker` instance be stored in a field, just to avoid (re)creating it over and over again on each call?